From 8d6a617ef159dcd8e07aaab528c76c9ed3540408 Mon Sep 17 00:00:00 2001
From: mbelchev <marianbelchev@gmail.com>
Date: Mon, 19 Jan 2015 22:22:38 +0200
Subject: [PATCH] Rspec for account class

---
 spec/models/account_spec.rb | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/spec/models/account_spec.rb b/spec/models/account_spec.rb
index e69de29..613b7ea 100644
--- a/spec/models/account_spec.rb
+++ b/spec/models/account_spec.rb
@@ -0,0 +1,29 @@
+require 'spec_helper'
+require 'yt/models/account'
+
+describe Yt::Account do
+	subject(:account) { Yt::Account.new }
+
+	describe '#upload_video' do
+		context 'return an Errno:ENOENT' do
+			it do
+				session = double('doubled session')
+				allow(account).to receive(:open).and_return true
+				allow(session).to receive(:upload_video).and_return Yt::Video
+				expect(account.upload_video('testPath', {:title => 'testTitle'})) rescue Errno::ENOENT
+			end
+		end
+	end
+
+	describe '#create_playlist' do
+		context 'creating playlist' do
+			it { expect(account.create_playlist('randomName')) rescue NoMethodError }
+		end
+	end
+
+	describe '#videos_params' do
+		context 'return all the videos owned by account' do
+			it { expect(account.videos_params).to eq :for_mine => true }
+		end
+	end
+end
\ No newline at end of file
-- 
1.9.1

