--- video_spec.rb	2015-01-12 21:45:27.152020647 +0200
+++ new.rb	2015-01-12 21:44:54.448022015 +0200
@@ -1,4 +1,3 @@
-require 'spec_helper'
 require 'yt/models/video'
 
 describe Yt::Video do
@@ -11,6 +10,48 @@
     end
   end
 
+  describe '#tags' do
+	context 'given tags' do
+		it {expect(video.snippet).to be 10}
+	end
+  end
+  
+  describe '#exists?' do
+	context 'is it really nil?' do
+		it {expect(video.id).to be nil}
+	end
+  end
+  
+ 	describe '#like' do
+		let(:attrs) { {id: 'video-id'} }
+
+		context 'given a not liked video' do
+			it ".like changes video rating" do
+				rating  = double("rating")
+				rating_new_value = :like
+				allow(video).to receive(:rating).and_return(rating)
+				allow(rating).to receive(:set).and_return(rating_new_value)
+				allow(rating).to receive(:rating).and_return(rating_new_value)
+				expect(video.like).to be true
+			end
+		end
+	end
+
+	describe '#dislike' do
+		let(:attrs) { {id: 'video-id'} }
+
+		context 'given a liked video' do
+			it ".dislike changes video ratings" do
+				rating  = double("rating")
+				rating_new_value = :dislike
+				allow(video).to receive(:rating).and_return(rating)
+				allow(rating).to receive(:set).and_return(rating_new_value)
+				allow(rating).to receive(:rating).and_return(rating_new_value)
+				expect(video.dislike).to be true
+			end
+		end
+	end
+
   describe '#statistics_set' do
     context 'given fetching a video returns statistics' do
       let(:attrs) { {statistics: {"viewCount"=>"202"}} }
