diff --git a/spec/models/playlist_spec.rb b/spec/models/playlist_spec.rb
index eaefa32..7e1ee3b 100644
--- a/spec/models/playlist_spec.rb
+++ b/spec/models/playlist_spec.rb
@@ -37,9 +37,16 @@ describe Yt::Playlist do
     it { expect(playlist.update title: 'new').to be true }
     it { expect{playlist.update title: 'new'}.to change{playlist.title} }
   end
-
+ 
   describe '#delete' do
     let(:attrs) { {id: 'PLSWYkYzOr'} }
+	it "deletes the playlist" do
+	options = nil
+	allow(playlist).to receive(:do_delete).and_yield
+	allow(playlist.delete).to receive(:id).and_return(nil)
+	expect(playlist.delete).to be true
+	end
+  
 
     context 'given an existing playlist' do
       before { expect(playlist).to receive(:do_delete).and_yield }
@@ -48,4 +55,4 @@ describe Yt::Playlist do
       it { expect{playlist.delete}.to change{playlist.exists?} }
     end
   end
-end
\ No newline at end of file
+end
