From 360d61c1b6dcc5afbfc31385762ccb78ba260d69 Mon Sep 17 00:00:00 2001
From: LuboYankov <ahhak123@gmail.com>
Date: Wed, 14 Jan 2015 17:12:49 +0200
Subject: [PATCH] Increasing subscription.rb coverage with 15%

---
 spec/models/subscription_spec.rb | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/spec/models/subscription_spec.rb b/spec/models/subscription_spec.rb
index eb8301c..f222f2c 100644
--- a/spec/models/subscription_spec.rb
+++ b/spec/models/subscription_spec.rb
@@ -27,4 +27,13 @@ describe Yt::Subscription do
       it { expect{subscription.delete}.to change{subscription.exists?} }
     end
   end
-end
\ No newline at end of file
+  
+  describe '#delete_params' do
+	let(:id) { 'CBl6OoF0BpiV'}
+		it 'delete parameters' do
+			expect(subscription.send(:delete_params)).to include :expected_response 
+			expect(subscription.send(:delete_params)).to include :path 
+			expect(subscription.send(:delete_params)).to include :method
+		end
+  end
+end
-- 
1.9.1

