diff --git a/spec/models/status_spec.rb b/spec/models/status_spec.rb
index 22a9185..bd52ebd 100644
--- a/spec/models/status_spec.rb
+++ b/spec/models/status_spec.rb
@@ -341,4 +341,68 @@ describe Yt::Status do
       it { expect(status).not_to have_public_stats_viewable }
     end
   end
-end
\ No newline at end of file
+
+  #subject(:status) { Yt::Status.new }
+
+  describe "#public?" do
+    it "public? returns true" do
+      status = Yt::Status.new
+      allow(status).to receive(:privacy_status).and_return('public')
+      expect(status.public?).to be true
+    end
+  end
+
+  describe "#embeddable" do
+    it "embedddable retruns true" do
+        status = Yt::Status.new
+      #embed = double("mocked var")
+      allow(status).to receive(:embeddable?).and_return(true)
+      expect(status.embeddable).to be true
+    end
+  end
+
+
+  describe "#public_stats_viewable" do
+    it "public_stats_viewable return true" do
+      status = Yt::Status.new
+      allow(status).to receive(:has_public_stats_viewable?).and_return(true)
+      expect(status.public_stats_viewable).to be true
+    end
+  end
+
+  describe "#video?" do
+    it "video? returns true" do
+      status = Yt::Status.new
+      upload_status = double("mocked upload_status")
+      allow(status).to receive(:upload_status).and_return(upload_status)
+      allow(upload_status).to receive(:present?).and_return(true)
+      expect(status.send(:video?)).to be true
+    end
+  end
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+end
