diff --git a/spec/models/video_spec.rb b/spec/models/video_spec.rb
index 84b087f..cc30a22 100644
--- a/spec/models/video_spec.rb
+++ b/spec/models/video_spec.rb
@@ -1,4 +1,4 @@
-require 'spec_helper'
+require_relative '../spec_helper'
 require 'yt/models/video'
 
 describe Yt::Video do
@@ -43,4 +43,40 @@ describe Yt::Video do
       it { expect{video.delete}.to change{video.exists?} }
     end
   end
-end
\ No newline at end of file
+
+
+	rating = nil
+	
+	describe '#like' do
+		let(:attrs) { {id: 'video-id'} }
+			it "liked video" do
+				rating = double('rating')
+				allow(video).to receive(:rating).and_return(rating)
+				allow(rating).to receive(:set).and_return(nil)
+			 	allow(rating).to receive(:rating).and_return(:like)
+			 	expect(video.like).to be true
+			end
+	end
+	
+	describe '#dislike' do
+		let(:attrs) { {id: 'video-id'} }
+			it "disliked video" do
+				rating = double('rating')
+				allow(video).to receive(:rating).and_return(rating)
+				allow(rating).to receive(:set).and_return(nil)
+				allow(rating).to receive(:rating).and_return(:dislike)
+				expect(video.dislike).to be true
+			end
+	end
+
+	describe '#unlike' do
+		let(:attrs) { {id: 'video-id'} }
+			it "unliked video" do
+				rating = double('rating')
+				allow(video).to receive(:rating).and_return(rating)
+				allow(rating).to receive(:set).and_return(nil)
+				allow(rating).to receive(:rating).and_return(:none)
+				expect(video.unlike).to be true
+			end
+	end
+end
