From 5272a80c9f5d3fe187b339ec1e4f88fa2bc8bdca Mon Sep 17 00:00:00 2001
From: Radoslav Kostadinov <radoslav97@abv.bg>
Date: Tue, 13 Jan 2015 01:44:38 +0200
Subject: [PATCH] few tests

---
 spec/models/video_spec.rb | 45 +++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 45 insertions(+)

diff --git a/spec/models/video_spec.rb b/spec/models/video_spec.rb
index 84b087f..2df4a18 100644
--- a/spec/models/video_spec.rb
+++ b/spec/models/video_spec.rb
@@ -43,4 +43,49 @@ describe Yt::Video do
       it { expect{video.delete}.to change{video.exists?} }
     end
   end
+   describe '#liked?' do
+   let(:attrs) {{id:'id'}}
+
+   it "liked? equals true" do
+    rating = double("rating")
+    allow(video).to receive(:rating).and_return(rating)
+    allow(rating).to receive(:rating).and_return(:like)
+    expect(video.liked?).to eql true
+  end
+end
+describe '#like' do
+ let(:attrs) {{id:'id'}}
+
+ it "like returns true" do
+  rating = double("rating")
+  allow(video).to receive(:rating).and_return(rating)
+  allow(rating).to receive(:rating).and_return(:like)
+  allow(rating).to receive(:set).and_return(:like)
+  expect(video.like).to be true
+end
+end
+
+describe '#unlike' do
+ let(:attrs) {{id:'id'}}
+
+ it "unlike returns true" do
+  rating = double("rating")
+  allow(video).to receive(:rating).and_return(rating)
+  allow(rating).to receive(:rating).and_return(:none)
+  allow(rating).to receive(:set).and_return(:none)
+  expect(video.unlike).to be true
+end
+end
+
+describe '#dislike' do
+ let(:attrs) {{id:'id'}}
+
+ it "dislike returns true" do
+  rating = double("rating")
+  allow(video).to receive(:rating).and_return(rating)
+  allow(rating).to receive(:rating).and_return(:dislike)
+  allow(rating).to receive(:set).and_return(:dislike)
+  expect(video.dislike).to be true
+ end
+end
 end
\ No newline at end of file
-- 
1.9.1

