diff --git a/spec/models/video_spec.rb b/spec/models/video_spec.rb
index 84b087f..e229315 100644
--- a/spec/models/video_spec.rb
+++ b/spec/models/video_spec.rb
@@ -43,4 +43,52 @@ describe Yt::Video do
       it { expect{video.delete}.to change{video.exists?} }
     end
   end
-end
\ No newline at end of file
+
+  describe '#likes&dislikes' do
+    let(:attrs) { {id: 'random-video-id'} }
+
+    rating = nil
+
+    before(:each) do
+      rating = double('rating')
+      allow(video).to receive(:rating).and_return(rating)
+      allow(rating).to receive(:set).and_return(nil)
+    end
+
+    describe '#like' do
+      context 'given a cool video' do
+        it "like video check" do
+          allow(rating).to receive(:rating).and_return(:like)
+          expect(video.like).to be true
+        end
+      end
+    end
+
+    describe '#why-not-liked' do
+      context 'given a not so cool video' do
+        it "not liked video check" do
+          allow(rating).to receive(:rating).and_return(:none)
+          expect(video.like).to be false
+        end
+      end
+    end
+
+    describe '#dislike' do
+      context 'given a so not cool video' do
+        it "dislike video check" do
+          allow(rating).to receive(:rating).and_return(:dislike)
+          expect(video.dislike).to be true
+        end
+      end
+    end
+
+    describe '#unlike' do
+      context 'given a dont like it anymore video' do
+        it "unlike video check" do
+          allow(rating).to receive(:rating).and_return(:none)
+          expect(video.unlike).to be true
+        end
+      end
+    end
+  end
+end
