diff --git a/spec/models/video_spec.rb b/spec/models/video_spec.rb
index 84b087f..b83b91a 100644
--- a/spec/models/video_spec.rb
+++ b/spec/models/video_spec.rb
@@ -43,4 +43,39 @@ describe Yt::Video do
       it { expect{video.delete}.to change{video.exists?} }
     end
   end
-end
\ No newline at end of file
+
+	describe '#like/dislike/unlike' do
+			let(:attrs) { {id: 'video-id'} }
+			rating = nil
+
+			describe "#like" do
+				it "like" do
+					rating = double('rating')
+					allow(video).to receive(:rating).and_return(rating)
+					allow(rating).to receive(:set).and_return(nil)
+					allow(rating).to receive(:rating).and_return(:like)
+					expect(video.like).to be true
+				end
+			end
+		
+			describe "#unlike" do
+				it "unlike" do
+					rating = double('rating')
+					allow(video).to receive(:rating).and_return(rating)
+					allow(rating).to receive(:set).and_return(nil)
+					allow(rating).to receive(:rating).and_return(:none)
+					expect(video.unlike).to be true
+				end
+			end
+
+			describe "#dislike" do
+				it "dislike" do
+					rating = double('rating')
+					allow(video).to receive(:rating).and_return(rating)
+					allow(rating).to receive(:set).and_return(nil)
+					allow(rating).to receive(:rating).and_return(:dislike)
+					expect(video.dislike).to be true
+				end
+			end
+	end
+end
